sycl/CMakeLists.txt

```diff
@@ -383,6 +383,12 @@ if(OpenCL_INSTALL_KHRONOS_ICD_LOADER AND TARGET OpenCL-ICD)
   list(APPEND SYCL_TOOLCHAIN_DEPLOY_COMPONENTS OpenCL-ICD)
 endif()
 
+# On Windows or for HIP AMD lld is also needed
+if(WIN32 OR ("hip" IN_LIST SYCL_ENABLE_PLUGINS AND "${SYCL_BUILD_PI_HIP_PLATFORM}" STREQUAL "AMD"))
```

Comment:
Can we simplify the condition here to following logic:
 - if lld is added to the list of enabled project then install it as part of sycl toolchain.
?

This should remove the duplication of the cmake configuration logic.

Reply:
Good point, simplified the condition as suggested and added an error for HIP if `lld` is missing